File: src/frontend/src/app/components/financeiro/titulo/model/titulo-dto.ts
export class TituloDTO {
  id?: string;
  tipo: string; // A_PAGAR, A_RECEBER
  status?: string; // ABERTO, PARCIAL, PAGO, CANCELADO, VENCIDO
  numeroDocumento?: string;
  descricao: string;

  // Relacionamentos
  pessoaId: string;
  pessoaNome?: string;
  unidadeNegocioId?: string;
  unidadeNegocioNome?: string;

  // Valores
  valorOriginal: number;
  valorPago?: number;
  valorDesconto?: number;
  valorJuros?: number;
  valorMulta?: number;
  saldo?: number;

  // Datas
  dataEmissao: Date;
  dataVencimento: Date;
  dataPagamento?: Date;

  observacoes?: string;

  // Parcelamento
  numeroParcela?: number;
  totalParcelas?: number;
  tituloOrigemId?: string;

  createdAt?: Date;
  updatedAt?: Date;
  createdBy?: string;
  updatedBy?: string;

  constructor(
    id?: string,
    tipo?: string,
    status?: string,
    numeroDocumento?: string,
    descricao?: string,
    pessoaId?: string,
    pessoaNome?: string,
    unidadeNegocioId?: string,
    unidadeNegocioNome?: string,
    valorOriginal?: number,
    valorPago?: number,
    valorDesconto?: number,
    valorJuros?: number,
    valorMulta?: number,
    saldo?: number,
    dataEmissao?: Date,
    dataVencimento?: Date,
    dataPagamento?: Date,
    observacoes?: string,
    numeroParcela?: number,
    totalParcelas?: number,
    tituloOrigemId?: string,
    createdAt?: Date,
    updatedAt?: Date,
    createdBy?: string,
    updatedBy?: string
  ) {
    this.id = id;
    this.tipo = tipo!;
    this.status = status;
    this.numeroDocumento = numeroDocumento;
    this.descricao = descricao!;
    this.pessoaId = pessoaId!;
    this.pessoaNome = pessoaNome;
    this.unidadeNegocioId = unidadeNegocioId;
    this.unidadeNegocioNome = unidadeNegocioNome;
    this.valorOriginal = valorOriginal!;
    this.valorPago = valorPago;
    this.valorDesconto = valorDesconto;
    this.valorJuros = valorJuros;
    this.valorMulta = valorMulta;
    this.saldo = saldo;
    this.dataEmissao = dataEmissao!;
    this.dataVencimento = dataVencimento!;
    this.dataPagamento = dataPagamento;
    this.observacoes = observacoes;
    this.numeroParcela = numeroParcela;
    this.totalParcelas = totalParcelas;
    this.tituloOrigemId = tituloOrigemId;
    this.createdAt = createdAt;
    this.updatedAt = updatedAt;
    this.createdBy = createdBy;
    this.updatedBy = updatedBy;
  }


}

File: src/backend/src/main/java/br/com/grupopipa/gestaointegrada/financeiro/titulo/TituloDTO.java
package br.com.grupopipa.gestaointegrada.financeiro.titulo;

import br.com.grupopipa.gestaointegrada.core.dto.DTO;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

@Getter
@Setter
@Builder
public class TituloDTO implements DTO {

    private UUID id;
    private String tipo; // A_PAGAR, A_RECEBER
    private String status; // ABERTO, PARCIAL, PAGO, CANCELADO, VENCIDO
    private String numeroDocumento;
    private String descricao;

    // Relacionamentos
    private UUID pessoaId;
    private String pessoaNome;
    private UUID tituloCategoriaId;
    private String tituloCategoriaNome;
    private UUID unidadeNegocioId;
    private String unidadeNegocioNome;

    // Valores
    private BigDecimal valorOriginal;
    private BigDecimal valorPago;
    private BigDecimal valorDesconto;
    private BigDecimal valorJuros;
    private BigDecimal valorMulta;
    private BigDecimal saldo; // Calculado

    // Datas
    private LocalDate dataEmissao;
    private LocalDate dataVencimento;
    private LocalDate dataPagamento;

    private String observacoes;

    // Parcelamento
    private Integer numeroParcela;
    private Integer totalParcelas;
    private UUID tituloOrigemId;

    // Setores para rateio
    private List<TituloSetorDTO> setores;

    // Rateio automático
    private Boolean rateioAutomatico;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private String createdBy;
    private String updatedBy;
}
